### DISCUSSION TITLE
using UCF101 dataset in Avalanche for Continual Learning

### DISCUSSION THREAD

**Discussion by askrinihad** [ContinualAI/avalanche#1584] [Q&A]
Hello,

I am attempting to build a continual learning model using the Avalanche library, specifically using the UCF101 dataset. However, I have encountered an issue while trying to execute the provided code. Here's the code snippet:
train_dataset = UCF101(ucf_data_dir, ucf_label_dir, frames_per_clip=frames_per_clip,
                       step_between_clips=step_between_clips, train=True, transform=torchvision.transforms.ToTensor())
test_dataset = UCF101(ucf_data_dir, ucf_label_dir, frames_per_clip=frames_per_clip,
                      step_between_clips=step_between_clips, train=False, transform=torchvision.transforms.ToTensor())

scenario = nc_benchmark(
    train_dataset, test_dataset, n_experiences=60, shuffle=True, seed=1234, task_labels=True
)
The error message I encountered is as follows: Unsupported dataset: must have a valid targets field or has to be a Tensor Dataset with at least 2 Tensors
I have tried various methods in Avalanche to address the issue, but none have worked. If you can offer any insight on how to solve this problem or provide an alternative approach to using UCF101 with Avalanche for continual learning, I would greatly appreciate it.
Thank you for your assistance.

**Comment by askrinihad** [comment]
Just to clarify, after digging into the avalanche code, UCF101 dataset does not contain the field `target` which is need by the method [`_select_targets`](https://github.com/ContinualAI/avalanche/blob/20f896fb973e4d4248c992c5b27bc7eecbc38ab4/avalanche/benchmarks/utils/classification_dataset.py#L899C5-L899C20).
